front/src/modules/ui/tables/hooks/useInitializeEntityTable.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react';
+import { useRecoilState } from 'recoil';
+
+import { TABLE_MIN_COLUMN_NUMBER_BECAUSE_OF_CHECKBOX_COLUMN } from '../constants';
+import { numberOfTableColumnsState } from '../states/numberOfTableColumnsState';
+import { numberOfTableRowsState } from '../states/numberOfTableRowsState';
+
+import { useResetTableRowSelection } from './useResetTableRowSelection';
+import { useSetSoftFocusPosition } from './useSetSoftFocusPosition';
+
+export type TableDimensions = {
+  numberOfRows: number;
+  numberOfColumns: number;
+};
+
+export function useInitializeEntityTable({
+  numberOfRows,
+  numberOfColumns,
+}: TableDimensions) {
+  const resetTableRowSelection = useResetTableRowSelection();
+
+  useEffect(() => {
+    resetTableRowSelection();
+  }, [resetTableRowSelection]);
+
+  const [, setNumberOfTableColumns] = useRecoilState(numberOfTableColumnsState);
+  const [, setNumberOfTableRows] = useRecoilState(numberOfTableRowsState);
+
+  useEffect(() => {
+    setNumberOfTableColumns(numberOfColumns);
```

Comment:
@lucasbordeau Is there a way to avoid this two `re-render` for component using this recoil state ?
I saw in recoil api that default value can be passed only in the state declaration, maybe something is possible with the `effects` ?

Reply:
Yes we can use a useRecoilCallback which mutates multiple atoms at the same time, but it feels like a bit overkill here no ? https://recoiljs.org/docs/api-reference/core/useRecoilCallback/

Reply:
I'm not sure if it's overkill, as this hook is actually generating 3 renders at mount on the components using the related recoil state

Reply:
Sure maybe we can optimize it later ? I can create an issue for that.

Reply:
For the moment I created a unique table dimension state so it's only one change for this part

Reply:
https://github.com/twentyhq/twenty/issues/463

Reply:
Perfect ! let's keep a track on that for later :)